tests/testthat/test-get_code.R

```diff
@@ -695,3 +695,94 @@ testthat::test_that("data() call is returned when data name is provided as a cha
+  testthat::it("starting with underscore is detected in code dependency", {
+    td <- teal_data() |>
+      within({
+        `_add_column_` <- function(lhs, rhs) cbind(lhs, rhs) # nolint: object_name.
+        iris_ds <- `_add_column_`(iris, data.frame(new_col = "new column"))
+      })
+
+    testthat::expect_identical(
+      get_code(td, datanames = "iris_ds"),
+      paste(
+        sep = "\n",
+        "`_add_column_` <- function(lhs, rhs) cbind(lhs, rhs)",
+        "iris_ds <- `_add_column_`(iris, data.frame(new_col = \"new column\"))"
+      )
+    )
+  })
+
+  testthat::it("with space character is detected in code dependency", {
+    td <- teal_data() |>
+      within({
+        `add column` <- function(lhs, rhs) cbind(lhs, rhs) # nolint: object_name.
+        iris_ds <- `add column`(iris, data.frame(new_col = "new column"))
+      })
+
+    testthat::expect_identical(
+      get_code(td, datanames = "iris_ds"),
+      paste(
+        sep = "\n",
+        "`add column` <- function(lhs, rhs) cbind(lhs, rhs)",
+        "iris_ds <- `add column`(iris, data.frame(new_col = \"new column\"))"
+      )
+    )
+  })
+
+  testthat::it("without special characters is cleaned and detected in code dependency", {
+    td <- teal_data() |>
+      within({
+        `add_column` <- function(lhs, rhs) cbind(lhs, rhs)
+        iris_ds <- `add_column`(iris, data.frame(new_col = "new column"))
+      })
+
+    testthat::expect_identical(
+      get_code(td, datanames = "iris_ds"),
+      paste(
+        sep = "\n",
+        "add_column <- function(lhs, rhs) cbind(lhs, rhs)",
+        "iris_ds <- add_column(iris, data.frame(new_col = \"new column\"))"
+      )
+    )
+  })
+
+  testthat::it("with non-native pipe used as function is detected code dependency", {
+    td <- teal_data() |>
+      within({
+        `%add_column%` <- function(lhs, rhs) cbind(lhs, rhs)
+        iris_ds <- `%add_column%`(iris, data.frame(new_col = "new column"))
+      })
+
+    # Note that the original code is changed to use the non-native pipe operator
+    # correctly.
+    testthat::expect_identical(
+      get_code(td, datanames = "iris_ds"),
+      paste(
+        sep = "\n",
+        "`%add_column%` <- function(lhs, rhs) cbind(lhs, rhs)",
+        "iris_ds <- iris %add_column% data.frame(new_col = \"new column\")"
+      )
+    )
+  })
+
+  testthat::it("with non-native pipe is detected code dependency", {
+    td <- teal_data() |>
+      within({
+        `%add_column%` <- function(lhs, rhs) cbind(lhs, rhs)
+        iris_ds <- iris %add_column% data.frame(new_col = "new column")
+      })
+
+    # Note that the original code is changed to use the non-native pipe operator
+    # correctly.
+    testthat::expect_identical(
+      get_code(td, datanames = "iris_ds"),
+      paste(
+        sep = "\n",
+        "`%add_column%` <- function(lhs, rhs) cbind(lhs, rhs)",
+        "iris_ds <- iris %add_column% data.frame(new_col = \"new column\")"
+      )
+    )
+  })
```

Comment:
There is a missing text proving that `get_code(x, names = "%add_column%"` returns correct output. I think `names` should be normalized as well as it is possible to use back-ticked symbol there as well.

Reply:
[See last 2 commits](https://github.com/insightsengineering/teal.data/pull/340/files/e70bd47e231f5fa04f44419119f386da21a560b4..c5ef914c248dbe64e9f446f65ec04b9677d5386f)